examples/aggregate_merging/src/projectors.rs

```diff
@@ -0,0 +1,153 @@
+// model - in this case, using queries which only update count_a when EventA is received, and count_b
+// when EventB is received, would be sufficient to guarantee soundness.
+#[async_trait]
+impl<T: Clone + Into<ProjectorEvent> + Send + Sync + Serialize + DeserializeOwned> SqliteProjector<T, CounterError>
+    for CounterProjector
+{
+    async fn project(
+        &self,
+        event: &StoreEvent<T>,
+        connection: &mut PoolConnection<Sqlite>,
+    ) -> Result<(), CounterError> {
+        let existing: Option<Counter> = Counter::by_id(event.aggregate_id, &mut *connection)
+            .await?
+            .map(|existing| existing);
+        let payload: ProjectorEvent = event.payload.clone().into();
+        match payload {
+            ProjectorEvent::A => match existing {
+                Some(counter) => Ok(Counter::update(
+                    event.aggregate_id,
+                    counter.count_a + 1,
+                    counter.count_b,
+                    &mut *connection,
+                )
+                .await?),
+                None => Ok(Counter::insert(event.aggregate_id, 1, 0, &mut *connection).await?),
+            },
+            ProjectorEvent::B => match existing {
+                Some(counter) => Ok(Counter::update(
+                    event.aggregate_id,
+                    counter.count_a,
+                    counter.count_b + 1,
+                    &mut *connection,
+                )
+                .await?),
+                None => Ok(Counter::insert(event.aggregate_id, 0, 1, &mut *connection).await?),
+            },
+        }
+    }
+}
+
+#[derive(sqlx::FromRow, Debug)]
+pub struct Counter {
+    pub counter_id: Uuid,
+    pub count_a: i32,
+    pub count_b: i32,
+}
+
+impl Counter {
+    pub async fn by_id(id: Uuid, executor: impl Executor<'_, Database = Sqlite>) -> Result<Option<Self>, sqlx::Error> {
+        sqlx::query_as::<_, Self>("SELECT * FROM counters WHERE counter_id = $1")
+            .bind(id)
+            .fetch_optional(executor)
+            .await
+    }
+
+    pub async fn insert(
+        id: Uuid,
+        count_a: i32,
+        count_b: i32,
+        executor: impl Executor<'_, Database = Sqlite>,
+    ) -> Result<(), sqlx::Error> {
+        sqlx::query_as::<_, Self>("INSERT INTO counters (counter_id, count_a, count_b) VALUES ($1, $2, $3)")
+            .bind(id)
+            .bind(count_a)
+            .bind(count_b)
+            .fetch_optional(executor)
+            .await
+            .map(|_| ())
+    }
+
+    pub async fn update(
+        id: Uuid,
+        count_a: i32,
+        count_b: i32,
+        executor: impl Executor<'_, Database = Sqlite>,
+    ) -> Result<(), sqlx::Error> {
+        sqlx::query_as::<_, Self>("UPDATE counters SET count_a = $2, count_b = $3 WHERE counter_id = $1")
+            .bind(id)
+            .bind(count_a)
+            .bind(count_b)
+            .fetch_optional(executor)
+            .await
+            .map(|_| ())
+    }
```

Comment:
maybe I would go with a more specific implementation. Instead of a generic update, maybe it's better to have explicit increaseA and increaseB functions (that uses update underneath). I think that the api would be better, especially in the projector implementation. WDYT?

Reply:
Changed update to take an enum describing the update to be done (CounterUpdate::A or CounterUpdate::B), to make it more explicit what's being done by the projector - is that alright?

Reply:
even better! :+1: